src/ResourceManager/DevSpaces/Commands.DevSpaces.Test/ScenarioTests/TestController.cs

```diff
@@ -0,0 +1,96 @@
+using ResourceManagementClient = Microsoft.Azure.Management.ResourceManager.ResourceManagementClient;
+using TestEnvironmentFactory = Microsoft.Rest.ClientRuntime.Azure.TestFramework.TestEnvironmentFactory;
+using TestUtilities = Microsoft.Rest.ClientRuntime.Azure.TestFramework.TestUtilities;
+
+namespace Microsoft.Azure.Commands.DevSpaces.Test.ScenarioTests
+{
+    public class TestController
+    {
+        private EnvironmentSetupHelper helper;
+
+        public TestController()
+        {
+            helper = new EnvironmentSetupHelper();
+        }
+
+        public static TestController NewInstance
+        {
+            get
+            {
+                return new TestController();
+            }
+        }
+
+        private void SetupManagementClients(MockContext context)
+        {
+            helper.SetupManagementClients(
+                context.GetServiceClient<ContainerServiceClient>(TestEnvironmentFactory.GetTestEnvironment()),
+                context.GetServiceClient<ResourceManagementClient>(TestEnvironmentFactory.GetTestEnvironment()),
+                context.GetServiceClient<InternalResourceManagementClient>(TestEnvironmentFactory.GetTestEnvironment()),
+                context.GetServiceClient<DevSpacesManagementClient>(TestEnvironmentFactory.GetTestEnvironment()));
+        }
+
+        public void RunPowerShellTest(params string[] scripts)
+        {
+            var callingClassType = TestUtilities.GetCallingClass(2);
+            var mockName = TestUtilities.GetCurrentMethodName(2);
+
+            Dictionary<string, string> d = new Dictionary<string, string>();
+            d.Add("Microsoft.Resources", null);
+            d.Add("Microsoft.Features", null);
+            d.Add("Microsoft.Authorization", null);
+            var providersToIgnore = new Dictionary<string, string>();
+            providersToIgnore.Add("Microsoft.Azure.Management.Resources.ResourceManagementClient", "2016-02-01");
+            HttpMockServer.Matcher = new PermissiveRecordMatcherWithApiExclusion(true, d, providersToIgnore);
+            HttpMockServer.RecordsDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "SessionRecords");
+            using (MockContext context = MockContext.Start(callingClassType, mockName))
+            {
+                SetupManagementClients(context);
+
+                var callingClassName = callingClassType
+                                        .Split(new[] { "." }, StringSplitOptions.RemoveEmptyEntries)
+                                        .Last();
+                helper.SetupEnvironment(AzureModule.AzureResourceManager);
+                helper.SetupModules(AzureModule.AzureResourceManager,
+                    
+                    "ScenarioTests\\" + callingClassName + ".ps1",
+                    "ScenarioTests\\Common.ps1",
+                    helper.RMProfileModule,
+                    helper.RMResourceModule,
+                    helper.GetRMModulePath(@"AzureRM.Tags.psd1"),
```

Comment:
Remove this once you take out the get-tag call

Reply:
Required